node/src/blockchain/blockchain_interface.rs

```diff
@@ -369,44 +422,174 @@ where
             .map_err(|e| BlockchainError::QueryFailed(e.to_string()))
             .wait()
     }
+}
 
-    fn send_transaction_tools<'a>(
-        &'a self,
-        fingerprint_request_recipient: &'a Recipient<PendingPayableFingerprint>,
-    ) -> Box<dyn SendTransactionToolsWrapper + 'a> {
-        Box::new(SendTransactionToolWrapperReal::new(
-            &self.web3,
-            fingerprint_request_recipient,
-        ))
-    }
+#[derive(Debug, PartialEq, Clone)]
+pub enum ProcessedPayableFallible {
+    Correct(PendingPayable),
+    Failure(RpcPayableFailure),
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct RpcPayableFailure {
+    pub rpc_error: Error,
+    pub recipient_wallet: Wallet,
+    pub hash: H256,
 }
 
+type HashAndAmountResult = Result<Vec<(H256, u128)>, PayableTransactionError>;
+
 impl<T> BlockchainInterfaceNonClandestine<T>
 where
-    T: Transport + Debug + 'static,
+    T: BatchTransport + Debug + 'static,
 {
     pub fn new(transport: T, event_loop_handle: EventLoopHandle, chain: Chain) -> Self {
-        let web3 = Web3::new(transport);
+        let web3 = Web3::new(transport.clone());
+        let batch_web3 = Web3::new(Batch::new(transport));
+        let batch_payable_tools = Box::new(BatchPayableToolsReal::<T>::default());
         let contract =
             Contract::from_json(web3.eth(), chain.rec().contract, CONTRACT_ABI.as_bytes())
                 .expect("Unable to initialize contract.");
+
         Self {
             logger: Logger::new("BlockchainInterface"),
             chain,
             _event_loop_handle: event_loop_handle,
             web3,
+            batch_web3,
+            batch_payable_tools,
             contract,
         }
     }
 
-    fn prepare_signed_transaction(
+    fn sign_and_register_multiple_payments(
+        &self,
+        consuming_wallet: &Wallet,
+        gas_price: u64,
+        pending_nonce: U256,
+        accounts: &[PayableAccount],
+    ) -> HashAndAmountResult {
+        let init: (HashAndAmountResult, Option<U256>) = (Ok(vec![]), Some(pending_nonce));
```

Comment:
You try launching a new idea useful for this or also other situations: We know how many items we will have in the vector when the operation is completed. 

So, because we're gonna use a mutable vector with pushing, let's just call `Vec::with_capacity(accounts.len())`.